src/test/java/ibm/jceplus/junit/base/BaseTestRSATypeCheckEnabled.java

```diff
@@ -51,7 +52,17 @@ protected void setUp() throws Exception {
     //
     public void testPrivateKeyEncrypt() throws Exception {
         try {
-            Cipher cp = Cipher.getInstance("RSA", providerName);
+            Cipher cp = null;
+            try {
+                cp = Cipher.getInstance("RSA", providerName);
+            } catch (NoSuchAlgorithmException nsae) {
+                if (providerName.equals("OpenJCEPlusFIPS")) {
```

Comment:
You can use the `provider.isFIPS()` to check instead of the name.

Reply:
Given the extensive patter of doing string comparision checks and the way in which the desired provider we want to use is passed around, for now lets keep this code as is.